pkg/plugin/watch.go

```diff
@@ -265,35 +265,24 @@ func (e *AutoscaleEnforcer) watchVMEvents(
 
 				if newVM.Status.PodName == "" {
 					logger.Info("Skipping update for VM because .status.podName is empty", util.VMNameFields(newVM))
-					return
-				}
-
-				if oldInfo.ScalingEnabled && !newInfo.ScalingEnabled {
+				} else if oldInfo.ScalingEnabled && !newInfo.ScalingEnabled {
 					logger.Info("Received update to disable autoscaling for VM", util.VMNameFields(newVM))
 					name := util.NamespacedName{Namespace: newInfo.Namespace, Name: newVM.Status.PodName}
 					callbacks.submitVMDisabledScaling(logger, name)
-				}
-
-				if (!oldInfo.ScalingEnabled || !newInfo.ScalingEnabled) && oldInfo.Using() != newInfo.Using() {
+				} else if (!oldInfo.ScalingEnabled || !newInfo.ScalingEnabled) && oldInfo.Using() != newInfo.Using() {
 					podName := util.NamespacedName{Namespace: newInfo.Namespace, Name: newVM.Status.PodName}
 					logger.Info("Received update changing usage for VM", zap.Object("old", oldInfo.Using()), zap.Object("new", newInfo.Using()))
 					callbacks.submitNonAutoscalingVmUsageChanged(logger, newInfo, podName.Name)
+				} else if oldVM.Status.PodName != newVM.Status.PodName {
+					// If the pod changed, then we're going to handle a deletion event for the old pod,
+					// plus creation event for the new pod. Don't worry about it - because all VM
+					// information comes from this watch.Store anyways, there's no possibility of missing
+					// an update.
+				} else if oldInfo.EqualScalingBounds(*newInfo) {
+					// If bounds didn't change, then no need to update
```

Comment:
This seems suspect; why doesn't that work in the old version?

Reply:
Good point.